File: app/src/main/java/fr/insapp/insapp/models/ImageSize.java
package fr.insapp.insapp.models;

import android.os.Parcelable;

import auto.parcelgson.AutoParcelGson;
import auto.parcelgson.gson.annotations.SerializedName;

/**
 * Created by thomas on 12/07/2017.
 */

@AutoParcelGson
public abstract class ImageSize implements Parcelable {

    @SerializedName("width")
    abstract int width();

    @SerializedName("height")
    abstract int height();

    static ImageSize create(int width, int height) {
        return new AutoParcelGson_ImageSize(width, height);
    }

    public int getWidth() {
        return width();
    }

    public int getHeight() {
        return height();
    }
}

File: app/src/main/java/fr/insapp/insapp/models/ImageSize.kt
package fr.insapp.insapp.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

/**
 * Created by thomas on 12/07/2017.
 * Kotlin rewrite on 30/08/2019.
 */

@Parcelize
data class ImageSize (
    @SerializedName("width") val width: Int,
    @SerializedName("height") val height: Int
): Parcelable
